string type
string color
---
bool success
string message
